src/server/internalServer.cpp

```diff
@@ -182,6 +181,7 @@ bool InternalServer::start() {
   auto server_start_time = std::chrono::system_clock::now().time_since_epoch();
   m_server_id = kiwix::to_string(server_start_time.count());
   m_library_id = m_server_id;
+  kiwix::fillLanguagesMap();
```

Comment:
This is still not the best way of filling the languages map only once. In particular it assumes that `libkiwix` will be used in a way where a `kiwix::Server` object is created only once. Do you know about the [Singleton](https://en.wikipedia.org/wiki/Singleton_pattern) design pattern and/or about [`std::call_once`](https://en.cppreference.com/w/cpp/thread/call_once)?

Reply:
I don't but learning what it is :+1: